src/three/TilesRenderer.js

```diff
@@ -28,7 +28,15 @@ const vecZ = new Vector3();
 const X_AXIS = new Vector3( 1, 0, 0 );
 const Y_AXIS = new Vector3( 0, 1, 0 );
 
-function emptyRaycast() {}
+function overridenRaycast( raycaster, intersects ) {
+
+	if ( ! this.userData.tilesRenderer.optimizeRaycast ) {
+
+		Object.getPrototypeOf( this ).raycast.call( this, raycaster, intersects );
+
+	}
+
+}
```

Comment:
Great! This is what I was thinking. However instead of putting the tiles renderer in userdata lets create a new function per tiles renderer instead that can be used for the objects (from what I can tell userdata is really intended for serializable data). So the function can be created in the TilesRenderer constructor:

```js
constructor( ... ) {

  // ...

  const tilesRenderer = this;
  this._overridenRaycast = function( raycaster, intersects ) {

    if ( tilesRenderer.optimizeRaycast ) {

      Object.getPrototypeOf( this ).raycast.call( this, raycaster, intersects );

    }

  };

}
```

And then we can use that function on each of the loaded objects:

```js
c.raycast = this._overridenRaycast;
```

Reply:
I think this is resolved now.